inbm/dispatcher-agent/dispatcher/source/constants.py

```diff
@@ -25,16 +26,16 @@ class SourceParameters:
 
 @dataclass(kw_only=True, frozen=True)
 class ApplicationAddSourceParameters:
-    gpg_key_uri: str
-    gpg_key_name: str
     file_name: str
     sources: list[str] = field(default_factory=lambda: [])
+    gpg_key_uri: Optional[str] = field(default=None)
+    gpg_key_name: Optional[str] = field(default=None)
```

Comment:
**Suggestion:** Use a more specific type hint for 'sources' to enforce that it is a list of strings rather than a list of any type. [best practice]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    from typing import List

    @dataclass(kw_only=True, frozen=True)
    class ApplicationAddSourceParameters:
        file_name: str
        sources: List[str] = field(default_factory=list)
        gpg_key_uri: Optional[str] = field(default=None)
        gpg_key_name: Optional[str] = field(default=None)
```

Reply:
Done